src/canonical/handler.js

```diff
@@ -540,8 +540,28 @@ export async function canonicalAuditRunner(baseURL, context, site) {
         const urlFormatChecks = validateCanonicalFormat(canonicalUrl, baseURL, log);
         checks.push(...urlFormatChecks);
 
-        const urlContentCheck = await validateCanonicalRecursively(canonicalUrl, log, options);
-        checks.push(...urlContentCheck);
+        // self-reference check
+        const selfRefCheck = canonicalTagChecks.find(
+          (c) => c.check === CANONICAL_CHECKS.CANONICAL_SELF_REFERENCED.check,
+        );
+        const isSelfReferenced = selfRefCheck?.success === true;
+
+        // if self-referenced - skip accessibility
+        if (isSelfReferenced) {
+          checks.push({
+            check: CANONICAL_CHECKS.CANONICAL_URL_STATUS_OK.check,
+            success: true,
+          });
+          checks.push({
+            check: CANONICAL_CHECKS.CANONICAL_URL_NO_REDIRECT.check,
```

Comment:
It seems that this line should be part of the previous `push` (after line 552).  Unless I am missing something? 🤔